src/mpid/ch4/netmod/ofi/ofi_win.h

```diff
@@ -320,9 +320,13 @@ static inline int MPIDI_NM_mpi_win_start(MPIR_Group * group, int assert, MPIR_Wi
     MPIDI_CH4U_ACCESS_EPOCH_CHECK_NONE(win, mpi_errno, goto fn_fail);
 
     MPIR_Group_add_ref(group);
+    if (assert & MPI_MODE_NOCHECK) {
+        MPIDI_CH4U_WIN(win, sync).assert_mode |= MPI_MODE_NOCHECK;
+        goto no_check;
```

Comment:
It looks like `MPIDI_CH4U_WIN(win, sync).assert_mode` is not reset properly. I would suggest reseting it at every epoch opening call. This test can catch this issue:
```
lockall(assert=nocheck)
unlockall()
lockall(assert=0) 
unlockall()
```